c++/triqs/gfs/functions/density.cpp

```diff
@@ -159,9 +160,9 @@ namespace triqs::gfs {
     for (int idx : range(0, res.shape()[0])) res(idx, idx) = dcomplex(0., imag(res(idx, idx)));
 
     res *= dcomplex(0., 1.) * dw / M_PI; // scale to density
-    res = 0.5 * (res + dagger(res));     // A = i( g - g^+ )
+    den = 0.5 * (res + dagger(res));     // A = i( g - g^+ )
 
-    return res;
+    return den;
```

Comment:
No need for the intermediate variable

```suggestion
    return 0.5 * (res + dagger(res));
```

Same applies below.

Maybe it also makes sense to issue a warning in case the input Green function violates hermiticity above a certain threshold?